zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -170,13 +172,21 @@ public void recordPathStat(Stat stat, OptionalLong notificationTime) {
       }
     }
   }
-
   protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long operationRetryTimeout,
       PathBasedZkSerializer zkSerializer, String monitorType, String monitorKey,
       String monitorInstanceName, boolean monitorRootPathOnly) {
+    this(null, zkConnection, connectionTimeout, operationRetryTimeout,
+        zkSerializer, monitorType, monitorKey,
+        monitorInstanceName, monitorRootPathOnly);
+  }
+
+  protected ZkClient(String realmKey, IZkConnection zkConnection, int connectionTimeout, long operationRetryTimeout,
```

Comment:
This is the native ZkClient. I don't think this is a good idea to touch the native ZkClient for the reasons stated above.